### PR TITLE
Mobile-Notifications —>Mobile

### PR DESCRIPTION
**Pull request by GGLozanov**
Implementation of the Mobile notifications feature using Firebase Cloud Messaging (FCM) and Cloud Functions (their implementations are located [here](https://github.com/katapultman/TUESPairs-Dev-Firebase-Functions) and [here](https://github.com/katapultman/TUESPairs-Prod-Firebase-Functions)), as well as DB support for the notifications collection (created/written inside the Cloud Functions and read/deleted in Mobile), along with many other small QoL changes and miscellaneous fixes.

There are other small bugfixes and small details left to be addressed regarding notification display, but they will be done after the localization feature.

Requested review!

### CODE REVIEW

**Review by msimov02 — APPROVED** [review]
LGTM!

File: Mobile/tues_pairs/lib/widgets/user_display/user_list.dart [hunk 1]
```diff
@@ -78,7 +78,7 @@ class _UserListState extends State<UserList> {
   int getListIndex(User user, {User currentUser}) {
     // TODO: Optimise this retarded ass fix
     logger.w('UserCard: onSkip user w/ id "' + currentUser.uid +
-        '" has skipped incorrect users and has triggered skip exception handler.'); // user removes bad
+        '" has skipped incorrect users and has tt\t\t[[triggered skip exception handler.'); // user removes bad
```
Comment (marianbitsov23) [inline Mobile/tues_pairs/lib/widgets/user_display/user_list.dart:81]: ??? typo

**Review by marianbitsov23 — APPROVED** [review]
Overall, well done both on the feature implementation + design, just fix the typing error i mentioned. Good job, looking forward to sync it with Web!

File: Mobile/tues_pairs/lib/widgets/user_display/user_list.dart (hunk 1 above)
Comment (GGLozanov) [reply Mobile/tues_pairs/lib/widgets/user_display/user_list.dart:81]: My computer had a stroke.

**Comment by GGLozanov** [comment]
Thanks! Will fix it, but since it's a log typo, it's nothing much. Changes approved.

**Pull request merged**